docs/decisions/0007-compile-common-theme-for-all-mfes.rst

```diff
@@ -0,0 +1,116 @@
+MFE-specific code, and build them separately.
+
+One thing that prevents this is that MFEs need these SCSS imports to get
+variables. In a number of cases these can be replaced with either
+Paragon/Bootstrap classes, or CSS variables. A separate ADR in `Paragon
+<https://github.com/openedx/paragon/pull/1388>`_ covers the situations where
+this isn't possible yet.
+
+Decision
+********
+
+Common cores styling that is to be applied to all MFEs should be part of a
+separate stylesheet and built separately from the MFE-specific stylesheet.
+
+Both these stylesheets will be included in each MFE, however the common
+stylesheet can be loaded from a common source for all MFEs.This will allowing
+for deploying the common theme stylesheet without redeploying the MFEs.
+
+The mechanism for loading this common stylesheet will need to be different from
+the mechanism for loading the MFE stylesheet. A JavaScript theme loader will
+contain the location of the SCSS file and will add this stylesheet to the
+document.
+
+The common theme package will be similar to the branding package, but can be
+deployed independently of MFEs. It will produce an CSS stylesheet, and a
+JavaScript loader that will load this stylesheet. When deploying MFEs, they can
+be provided a configuration variable that contains the location where this
+theme loader is deployed.
+
+The frontend-platform repo can host code that simplifies using themes. It can
+check if a theme loader is configured, and if so, dynamically load the theme
+from there. Otherwise, it can continue using the existing mechanism of building
+and deploying the theme for each MFE for backwards-compatibility. It can also
+provide hooks for loading the theme and potentially/eventually also selecting
+a theme.
+
+Consequences
+************
+
+With a single common stylesheet for all MFE, they should load quicker since the
+stylesheet can be cached once for all MFEs. It also makes theme rebuilds much
+quicker, and eliminates the need for rebuilding each MFE for minor change to
+the theme.
+
+By continuing to support the existing theming mechanism, the current mechanism
+for theming can continue working for those who need it.
+
+Since there is a JavaScript-based loader involved, there is potential for
+flashing of unstyled content during loading, however this can be avoided by
+making the theme loader part of the init process so it runs before the app is
+considered ready.
+
+Rejected Alternatives
+*********************
+
+- Loading the common CSS directly instead of with a JavaScript based loader
+  Adding a loader eliminates caching issues. Currently the CSS files include
+  the hash of the CSS code in the file name, so that with each redeployment the
+  file names change, ensuring that a file is cached as long as it's used but not
+  loaded any more once the file contents change.
+
+  Since the common theme needs to have a fixed name, we can't rely on changing
```

Comment:
This confused me at first. I think what you're saying is "Without using a JavaScript based loader on the client side, and without support for pre-rendering our MFEs on the server side, we would need to use a fixed URL/filename for the common CSS file, which would cause problems with caching that file."

Because in your actual ADR, the common theme does not need to have a fixed name ("now it can be a file with
  the hash in its name").